### DISCUSSION TITLE
Database schema is not in sync with your migration history

### DISCUSSION THREAD

**Discussion by david8801** [prisma/prisma#7708] [Q&A] [answered]
Want to add new migration, so I enter a migration command in my IDE terminal:
npx prisma migrate dev --name task_comments

I also have a digital ocean server running, where I pull my changes and restart server to have my changes.
The problem is, when I create migration from my IDE, I sometimes get a migration sync error on my digital ocean terminal, when trying to create another migration:

✔ - The migration `20210603142051_generated` failed.
- Drift detected: Your database schema is not in sync with your migration history.
- The following migration(s) are applied to the database but missing from the local migrations directory: 20210531213316_category, 20210603092559_random, 20210603142051_generated

We need to reset the PostgreSQL database "defaultdb" at "my-db-link".
Do you want to continue? All data will be lost.

Also, if I create migration on digital ocean terminal - it creates migration, but I get the same error in my IDE, when trying to add another migration from IDE.

What scares me is that every time I get this error - I need to clear my DB, which works for now, cos I'm currently on developer server, but while on production - I can't just clear DB.

Isn't there any option to pull last version of migrations from DB ?

I'm mainly a front end dev, so I may not fully understand how migrations work, but I would be very greatful for any help.

Also, where should I create migrations from, my IDE or digital ocean terminal, it there any difference ?

**Comment by tomhoule** [comment, marked as answer]
Hi @david8801 — it's normal for your development environments to get out of sync. If you do not plan on keeping and committing migrations while iterating in development on your DO database, `prisma db push` is a good alternative. The resets are only occurring with `prisma migrate dev`, when you want to migrate your production environment, you should use `prisma migrate deploy`, which will only apply migrations, and *never* reset your database.

**Reply by tomhoule** [reply]
Dev environments getting out of sync is often happening just because you are switching git branches or pulling new commits with migrations. The output of `migrate dev` is currently maybe a bit scary, but that's the expected, normal workflow.

**Reply by david8801** [reply]
Hi @tomhoule, thank you very much for explanation, got it now

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
